Sources/PipeableSDK/PageLoadState.swift

```diff
@@ -56,6 +60,10 @@ class PageLoadState {
 
             let removeListener = self.subscribeToLoadStateChange { state, url, error in
                 if let error = error {
+                    if ignoreNavigationErrors {
```

Comment:
Just to make sure I understand this, suppose a user wants to wait for a url `foo.com` with load state `load`, and they try to navigate to `foo.com` but `foo.com` does not exist, then in this case the `waitForURL` will block forever correct?

Reply:
It will block until the time out expires, yes.

I guess we could change it such that if the error is for a matching URL, then we propagate it. That makes sense, I think.

Reply:
Let me check what playwright handles that.

Reply:
Hmmmmm......

Playwright actually works like how pipeable works without this PR.

That is, if you have any error while waitForURL(), it just cancels it.

If we do that, we need a new way to figure out how to wait for an agent to login.

Reply:
Oh right, forgot about the timeout for `waitForURL`, makes sense. I actually think the behavior you have here matches what playwright does, iiuc here is code to replicate in playwright:

```
await page.goto("http://localhost:3000/load_latency/0/index.html");

    // start async waiting for a url 
    page.waitForURL("http://localhost:3000/load_latency/0/non_existing_page.html")
    .then(() => {
        console.log("URL changed to expected URL");
    })
    .catch((e) => {
        console.log("Error: ", e);
    });

    try {
        await page.goto("http://localhost:3000/load_latency/0/non_existing_page.html");
    } catch (e) {
        console.log("Good error: ", e);
    }

    // now navigate to the correct page
    await page.goto("http://localhost:3000/load_latency/0/another.html");
  ```
  and in this case we just print out `URL changed to expected URL` with no errors

Reply:
@jlozano what do you get with that code?

I get the "Error" case.

Reply:
@pdobrev ya sorry I get that too, my original code was sending the request to a route that didn't exist on the server so we just didn't get a response, once I switched it to a non existing server I get an error from the `goto` to the non existing server and an error in the `waitForURL` on a non existing server

Reply:
I think one easy option would be to just expose the `ignoreNavigationErrors` to developers in the API and set the default to `false` so the default behavior matches playwright but we can keep doing the hack for logins by setting `ignoreNavigationErrors` to true

Reply:
So the correct full code I was using is:

```
await page.goto("http://localhost:3000/load_latency/0/index.html");

    // start async waiting for a url 
    page.waitForURL("http://localhost:3000/load_latency/0/another.html")
    .then(() => {
        console.log("URL changed to expected URL");
    })
    .catch((e) => {
        console.log("Error from waitForURL: ", e);
    });

    try {
        // navigate to a non-existing server so we get an error
        await page.goto("http://localhost:3001/foo/non_existing_page_2.html");
    } catch (e) {
        console.log("Error from goto: ", e);
    }

    // now navigate to the correct page
    await page.goto("http://localhost:3000/load_latency/0/another.html");
 ```
 
And in this case we get `Error on goto` and `Error on waitForURL` so like you said the current behavior of Pipeable matches Playwright and this PR deviates.

Sorry for the confusion!